app/src/test/java/tech/ula/utils/ValidationUtilityTest.kt

```diff
@@ -0,0 +1,229 @@
+
+    @Before
+    fun setup() {
+        validationUtility = ValidationUtility(resources)
+
+        val blacklistedUsernames = arrayOf(
+                "adm",
+                "audio",
+                "autologin",
+                "backup",
+                "bin",
+                "cdrom",
+                "daemon",
+                "dbus",
+                "dialout",
+                "dip",
+                "disk",
+                "fax",
+                "floppy",
+                "ftp",
+                "games",
+                "gnats",
+                "http",
+                "input",
+                "irc",
+                "kmem",
+                "kvm",
+                "list",
+                "locate",
+                "lock",
+                "log",
+                "lp",
+                "lpadmin",
+                "mail",
+                "mem",
+                "netdev",
+                "network",
+                "news",
+                "nobody",
+                "nogroup",
+                "operator",
+                "optical",
+                "plugdev",
+                "postgres",
+                "power",
+                "proc",
+                "proxy",
+                "rfkill",
+                "root",
+                "sambashare",
+                "sasl",
+                "scanner",
+                "shadow",
+                "smmsp",
+                "src",
+                "staff",
+                "storage",
+                "sudo",
+                "sync",
+                "sys",
+                "systemd-journal",
+                "tape",
+                "tty",
+                "users",
+                "utmp",
+                "uucp",
+                "uuidd",
+                "vboxusers",
+                "video",
+                "voice",
+                "wheel",
+                "www-data"
+        )
+
+        whenever(resources.getStringArray(R.array.blacklisted_usernames))
+                .thenReturn(arrayOf("adm", "audio"))
+        whenever(resources.getString(R.string.error_empty_field))
+                .thenReturn("Each field must be entered!")
+        whenever(resources.getString(R.string.error_username_invalid_characters))
+                .thenReturn("Characters in username must be 1 to 30 characters with lowercase letters and/or numbers.")
+        whenever(resources.getStringArray(R.array.blacklisted_usernames))
+                .thenReturn(blacklistedUsernames)
+        whenever(resources.getString(R.string.error_password_invalid))
+                .thenReturn("Password has invalid characters")
+        whenever(resources.getString(R.string.error_vnc_password_invalid))
+                .thenReturn("VNC Password has invalid characters")
+        whenever(resources.getString(R.string.error_vnc_password_length_incorrect))
+                .thenReturn("VNC Password must be between 6 to 8 characters!")
+    }
+
+    @Test
+    fun `Username is empty`() {
```

Comment:
I think these test names could stand to be at least a little more verbose. Something like `Validation fails appropriately if username is empty`.

Reply:
Updated all test function names